### ISSUE TITLE
SA1002 doesn't handle inline comments well

### ISSUE THREAD

**Issue by Tinister** [DotNetAnalyzers/StyleCopAnalyzers#1426]
Example:

    public static int GetBitPattern(PType ptype)
    {
        return IsSpecial(ptype) ? 0x11001 /*legacy*/ : 0x100001 /*since april 2014*/;
    }

It wants to remove the space *before* the inline comment (starting with `/*since`).

**Comment by sharwell** [comment]
:bulb: This method should probably be rewritten.

```csharp
public static int GetBitPattern(PType ptype)
{
    // Since April, 2014
    const int CurrentBitPattern = 0x100001;

    // Before April, 2014
    const int LegacyBitPattern = 0x11001;

    return IsSpecial(ptype) ? LegacyBitPattern : CurrentBitPattern;
}
```

**Comment by Tinister** [comment]
Sure, but if teams wanted to discourage that behavior wouldn't that be more appropriate for a separate "don't use delimited comments in the middle of a line of code" rule instead of having rules like SA1002 misflag in those situations?

**Comment by sharwell** [comment]
I think it makes sense to not report SA1002 for this specific example shown above. I would expect it to be reported if a space appeared between the comment and the `;`.

@Noryoko what do you think? Should we only look for immediately preceding spaces?

@oatkins What does StyleCop "classic" do for this?

**Comment by Noryoko** [comment]
I agree. We should only check immediately preceding spaces. :+1:

**Comment by oatkins** [comment]
@sharwell I was (pleasantly) surprised to find that classic SC doesn't report anything for the example at the top of this file. You were correct in thinking that you'd only get an error if you were to insert a space before the semi-colon.
